pkgs/nix-tools/darwin-rebuild.sh

```diff
@@ -9,11 +9,13 @@ showSyntax() {
   echo "               [--list-generations] [{--profile-name | -p} name] [--rollback]" >&2
   echo "               [{--switch-generation | -G} generation] [--verbose...] [-v...]" >&2
   echo "               [-Q] [{--max-jobs | -j} number] [--cores number] [--dry-run]" >&2
-  echo "               [--keep-going] [-k] [--keep-failed] [-K] [--fallback] [--show-trace]" >&2
+  echo "               [--keep-going | -k] [--keep-failed | -K] [--fallback] [--show-trace]" >&2
+  echo "               [--print-build-logs | -L]" >&2
   echo "               [-I path] [--option name value] [--arg name value] [--argstr name value]" >&2
-  echo "               [--flake flake] [--no-flake] [--update-input input flake] [--impure]" >&2
-  echo "               [--recreate-lock-file] [--no-update-lock-file] [--refresh]" >&2
-  echo "               [--offline] [--substituters substituters-list] ..." >&2
+  echo "               [--no-flake | [--flake flake] [--no-registries] [--commit-lock-file]" >&2
+  echo "               [--no-update-lock-file] [--no-write-lock-file] [--recreate-lock-file]" >&2
+  echo "               [--override-input input flake] [--update-input input]] [--impure]" >&2
+  echo "               [--refresh] [--offline] [--substituters substituters-list] ..." >&2
```

Comment:
```suggestion
  echo "               [--refresh] [--offline] [--substituters substituters-list] ...]" >&2
```

Can you add a closing square bracket here for the flake options section?

Reply:
The closing square bracket is currently after `[--update-input input]`.

I'm not familiar with the

* `--impure`
* `--refresh`
* `--offline`
* `--substituters`

`nix build` flags but I believe they are not flake-specific?

Reply:
`--offline` and `--refresh` are `nix build` only (running `nix-build` with them outputs `error: unrecognised flag`)

My mistake I didn't see the closing square bracket, could you reorder the flags so that those flags are with the rest of the flake specific flags?

It might also be a good idea to add start a new line after the closing square bracket for the flake flags section

Reply:
Why if `--offline` treated as a build/metadata flag, but `--refresh` as a lock flag?

```
    --show-trace|--keep-going|--keep-failed|--verbose|-v|-vv|-vvv|-vvvv|-vvvvv|--fallback|--offline)
      extraMetadataFlags+=("$i")
      extraBuildFlags+=("$i")
      ;;
```
```
    -L|-vL|--print-build-logs|--impure|--recreate-lock-file|--no-update-lock-file|--no-write-lock-file|--no-registries|--commit-lock-file|--refresh)
      extraLockFlags+=("$i")
      ;;
```

Reply:
I note that the `nix build` `--no-registries` option is deprecated in favor of `--no-use-registries`.

Reply:
`--offline` and `--refresh` are now listed as flake options.

If you would like to make further changes, feel free to make the edit yourself.
You have permissions to make changes to the source branch.